src/OpenTelemetry.Extensions.Enrichment.AspNetCore/ConfigureAspNetCoreInstrumentationOptions.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="ConfigureAspNetCoreInstrumentationOptions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Extensions.Enrichment.AspNetCore;
+using OpenTelemetry.Instrumentation.AspNetCore;
+
+namespace Microsoft.R9.Extensions.HttpClient.Tracing.Internal;
+
+#pragma warning disable CA1812 // Class is instantiated through dependency injection
+internal sealed class ConfigureAspNetCoreInstrumentationOptions : IConfigureOptions<AspNetCoreInstrumentationOptions>
+#pragma warning restore CA1812 // Class is instantiated through dependency injection
+{
+    private readonly AspNetCoreTraceEnrichmentProcessor processor;
+
+    public ConfigureAspNetCoreInstrumentationOptions(AspNetCoreTraceEnrichmentProcessor processor)
+    {
+        this.processor = processor;
+    }
+
+    public void Configure(AspNetCoreInstrumentationOptions options)
+    {
+        options.EnrichWithHttpRequest = this.processor.EnrichWithHttpRequest;
```

Comment:
what if user already has EnrichWithHttpRequest? this will overwrite it right?

Reply:
Yes. So here is a limitation - you can't use both API options.

Reply:
Same with https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/949#discussion_r1114699978, we need to discuss the value adds of the aspnet enrichers over existing enrichers.

Reply:
I already mentioned the values it brings in the #949 comment. Please check it out.

Reply:
Resolving as AspNetCore enrichment is removed from this PR